dotnet/src/Agents/Abstractions/AgentChatSerializer.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Agents.Serialization;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Able to serialize and deserialize an <see cref="AgentChat"/>.
+/// </summary>
+public sealed class AgentChatSerializer
+{
+    private readonly AgentChatState _state;
+
+    private static readonly JsonSerializerOptions s_defaultOptions =
```

Comment:
Is there any way for a developer to override this or configure it? It could be needed for annoyances like known polymorphic types etc.

Reply:
Not currently.  Good point.  I'm curious if the _Kernel_ content types have considered unknown subclasses.

Reply:
Updated (found this https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/ChatCompletion/ChatHistorySerialization.cs#L61)